internal/provider/kubernetes/kubernetes.go

```diff
@@ -73,6 +79,13 @@ func New(cfg *rest.Config, svr *config.Server, resources *message.ProviderResour
 		return nil, fmt.Errorf("unable to set up ready check: %w", err)
 	}
 
+	// Emit elected & continue with deployment of infra resources
+	go func() {
+		if <-mgr.Elected(); true {
```

Comment:
```suggestion
		if <-mgr.Elected() {
```

Reply:
Switched to using <-mgr.Elected(), given that it's a channel of type struct{}.